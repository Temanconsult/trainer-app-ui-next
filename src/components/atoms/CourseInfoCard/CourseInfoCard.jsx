import React from 'react'
import PropTypes from 'prop-types'
import { Card } from 'react-bootstrap'

const CourseInfoCard = ({ course }) => {
  return (
    <Card
      className="mb-4 mr-md-2 mr-lg-4 mr-sm-0"
      text={course.bg === 'light' ? 'dark' : 'white'}
      bg={course.bg}
      style={{ maxWidth: '30rem' }}
    >
      <Card.Body>
        <Card.Img variant="top" src={course.image} />
        <Card.Title>{course.title}</Card.Title>
        <Card.Text>{course.body}</Card.Text>
      </Card.Body>
    </Card>
  )
}

CourseInfoCard.propTypes = {
  course: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    bg: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
}

export default CourseInfoCard
