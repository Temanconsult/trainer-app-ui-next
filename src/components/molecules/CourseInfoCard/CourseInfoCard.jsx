import React from 'react'
import PropTypes from 'prop-types'
import { Card } from 'react-bootstrap'
import image from '../../../assets/images/pmp.svg'

const CourseInfoCard = ({ course }) => {
  const bg = course.bg || 'light'
  const imageIcon = course.image || image
  return (
    <Card
      className="mb-4 mr-md-2 mr-lg-4 mr-sm-0"
      text={bg === 'light' ? 'dark' : 'white'}
      bg={bg}
      style={{ maxWidth: '30rem' }}
    >
      <Card.Body>
        <Card.Img variant="top" src={imageIcon} />
        <Card.Title>{course.name}</Card.Title>
        <Card.Text>{course.description}</Card.Text>
      </Card.Body>
    </Card>
  )
}

CourseInfoCard.propTypes = {
  course: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    bg: PropTypes.string,
    image: PropTypes.string,
  }),
}

CourseInfoCard.defaultProps = {
  course: {
    name: '',
    description: '',
    bg: 'light',
    image,
  },
}

export default CourseInfoCard
