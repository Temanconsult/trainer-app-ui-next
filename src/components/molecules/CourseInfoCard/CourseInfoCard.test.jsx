import React from 'react'
import { render } from '@testing-library/react'
import CourseInfoCard from '.'

describe('CourseInfoCard', () => {
  it('creates course info card', () => {
    const { asFragment } = render(
      <CourseInfoCard
        course={{
          title: 'Cloud',
          body:
            'Cloud computing is the on-demand availability of computer system resources, especially data storage and computing power, without direct active management by the user.',
          image: 'devops',
          bg: 'light',
        }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
