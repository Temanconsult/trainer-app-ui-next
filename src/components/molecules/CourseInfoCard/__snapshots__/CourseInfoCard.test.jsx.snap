// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseInfoCard creates course info card 1`] = `
<DocumentFragment>
  <div
    class="mb-4 mr-md-2 mr-lg-4 mr-sm-0 card bg-light text-dark"
    style="max-width: 30rem;"
  >
    <div
      class="card-body"
    >
      <img
        class="card-img-top"
        src="devops"
      />
      <div
        class="card-title h5"
      />
      <p
        class="card-text"
      />
    </div>
  </div>
</DocumentFragment>
`;
